module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -542,6 +542,42 @@ protected function fetchUserWithCql($query)
         return count($json->users) === 1 ? $json->users[0] : null;
     }
 
+    /**
+     * Helper function to retrieve paged results from FOLIO API
+     *
+     * @param string $responseKey Key containing values to collect in response
+     * @param string $interface   FOLIO api interface to call
+     * @param array  $query       CQL query
+     *
+     * @return array
+     */
+    protected function getPagedResults($responseKey, $interface, $query = [])
+    {
+        $fullResponse = [];
+        // set artificially low to test
+        $limit = 1000;
+        $offset = 0;
+
+        do {
+            $combinedQuery = compact('query', 'offset', 'limit');
+            $response = $this->makeRequest(
+                'GET',
+                $interface,
+                $combinedQuery
+            );
+            $json = json_decode($response->getBody());
+            $total = $json->totalRecords ?? 0;
+            $preCount = count($fullResponse);
+            foreach ($json->$responseKey ?? [] as $item) {
+                $fullResponse[] = $item ?? '';
```

Comment:
As you suggested in Slack, just putting a `yield` here and turning this function into a generator would probably improve performance.

Reply:
Turned out to be a little tricky--when the yield keyword was hit the loop started from the beginning and the rest of the logic was not evaluated. I refactored to get all the logic above the yield keyword. Hopefully not too clumsily. I do think having a generator might this easier to use in other functions. Might not be a big performance payoff though.